src/SFML/Audio/SoundFileWriterOpus.cpp

```diff
@@ -0,0 +1,259 @@
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Audio/SoundFileWriterOpus.hpp>
+#include <SFML/System/Err.hpp>
+#include <algorithm>
+#include <cstring>
+#include <ctime>
+#include <cassert>
+
+
+// Anonymous namespace
+namespace
+{
+    // Make sure to write int into buffer little endian
+    void writeUint32(unsigned char* buffer, sf::Uint32 value)
+    {
+        buffer[0] = static_cast<unsigned char>(value & 0x000000FF);
+        buffer[1] = static_cast<unsigned char>((value & 0x0000FF00) >> 8);
+        buffer[2] = static_cast<unsigned char>((value & 0x00FF0000) >> 16);
+        buffer[3] = static_cast<unsigned char>((value & 0xFF000000) >> 24);
+    }
+}
+
+namespace sf
+{
+namespace priv
+{
+////////////////////////////////////////////////////////////
+bool SoundFileWriterOpus::check(const std::string& filename)
+{
+    std::string extension = filename.substr(filename.find_last_of(".") + 1);
+    std::transform(extension.begin(), extension.end(), extension.begin(), ::tolower);
+
+    return extension == "opus";
+}
+
+
+////////////////////////////////////////////////////////////
+SoundFileWriterOpus::SoundFileWriterOpus() :
+m_channelCount (0),
+m_file         (),
+m_ogg          (),
+m_opus         (NULL),
+m_packageNumber(0)
+{
+}
+
+
+////////////////////////////////////////////////////////////
+SoundFileWriterOpus::~SoundFileWriterOpus()
+{
+    close();
+}
+
+
+////////////////////////////////////////////////////////////
+bool SoundFileWriterOpus::open(const std::string& filename, unsigned int sampleRate, unsigned int channelCount)
+{
+    // Save the channel count
+    m_channelCount = channelCount;
+    m_sampleRate = sampleRate;
+    std::srand(std::time(0));
+
+    // Initialize the ogg/opus stream
+    if (ogg_stream_init(&m_ogg, std::rand()) == -1)
+    {
+        err() << "Stream init of ogg/opus file \"" << filename << "\" failed" << std::endl;
+        close();
+        return false;
+    }
+    int status = 0;
```

Comment:
Is there no constant to denote an invalid status value? I'm asking because it may be possible that `OPUS_OK` can be defined as 0. Also, it would make code more clear (`0` is a magic constant here).